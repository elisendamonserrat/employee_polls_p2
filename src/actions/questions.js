export const RECEIVE_QUESTIONS = "RECEIVE_QUESTIONS";
export const ADD_QUESTION = "ADD_QUESTION";
export const VOTE_QUESTIONS = "VOTE_QUESTIONS";

export function receiveQuestions(questions) {
  return {
    type: RECEIVE_QUESTIONS,
    questions,
  };
}

export function addQuestion(question) {
  return {
    type: ADD_QUESTION,
    question,
  };
}
