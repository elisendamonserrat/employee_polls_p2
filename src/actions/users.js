import { _getUsers } from "../utils/_DATA";

export const RECEIVE_USERS = "RECEIVE_USERS";

export function receiveUsers(users) {
  return {
    type: RECEIVE_USERS,
    users,
  };
}

export function handleRecievUsers() {
  return (dispatch) => {
    return _getUsers()
      .then((users) => {
        dispatch(receiveUsers(users));
      })
      .catch((error) => console.log("Error: ", error.message));
  };
}
