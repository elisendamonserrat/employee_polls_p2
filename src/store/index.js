import reducer from "../reducers/index";
import { createStore } from "redux";
import middleware from "../middleware";

const store = createStore(reducer, middleware);

export default store;
