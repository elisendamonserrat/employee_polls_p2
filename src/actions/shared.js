import { _getQuestions, _getUsers, _saveQuestion } from "../utils/_DATA";
import { receiveQuestions, addQuestion } from "./questions";
import { receiveUsers } from "./users";
import { setAuthedUser } from "./authedUser";
import { showLoading, hideLoading } from "react-redux-loading-bar";

export function handleInitialData() {
  return (dispatch) => {
    dispatch(showLoading());
    return Promise.all([_getQuestions(), _getUsers()])
      .then(([questions, users]) => {
        dispatch(receiveQuestions(questions));
        dispatch(receiveUsers(users));
        dispatch(setAuthedUser(null));
        dispatch(hideLoading());
      })
      .catch((error) => {
        console.log("Error: ", error.message);
        dispatch(hideLoading());
      });
  };
}

// ASYNC ACTION THAT WILL UPDATE THE QUESTIONS AND USERS STATE
export function handleAddQuestion(question) {
  return (dispatch, getState) => {
    const { authedUser } = getState();
    dispatch(showLoading());
    return _saveQuestion({
      optionOneText: question.optionOne,
      optionTwoText: question.optionTwo,
      author: authedUser,
    })
      .then((question) => {
        dispatch(addQuestion(question));
      })
      .then(() => dispatch(hideLoading()))
      .catch((error) => console.log("Error: ", error));
  };
}
