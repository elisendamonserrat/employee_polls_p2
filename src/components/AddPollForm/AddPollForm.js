import React from "react";
import { useState } from "react";
import { useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";
import { handleAddQuestion } from "../../actions/shared";

const defaultPollState = {
  optionOne: "",
  optionTwo: "",
};

const AddPollForm = () => {
  const [newPoll, setNewPoll] = useState({
    optionOne: "",
    optionTwo: "",
  });
  const dispatch = useDispatch();
  const navigate = useNavigate();

  const handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;

    setNewPoll((state) => ({
      ...state,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    dispatch(handleAddQuestion(newPoll));
    setNewPoll(defaultPollState);
    navigate("/");
  };

  return (
    <>
      <form
        className="bg-white border shadow-md rounded px-8 pt-8 pb-8 mb-4 mt-8 text-left"
        onSubmit={handleSubmit}
      >
        <p className="subtitle mb-4 font-bold">Would You Rather...?</p>
        <div className="mb-4">
          <label className="block text-sm font-bold mb-2" htmlFor="optionOne">
            First Option
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
            id="optionOne"
            name="optionOne"
            type="text"
            placeholder="Write here your first option"
            value={newPoll.optionOne}
            onChange={handleChange}
            data-testid="optionOne"
          />
        </div>
        <div className="mb-6">
          <label className="block text-sm font-bold mb-2" htmlFor="optionTwo">
            Second Option
          </label>
          <input
            className="shadow appearance-none border rounded w-full py-2 px-3 mb-3 leading-tight focus:outline-none focus:shadow-outline"
            id="optionTwo"
            type="text"
            name="optionTwo"
            placeholder="Write here your second option"
            value={newPoll.optionTwo}
            onChange={handleChange}
            data-testid="optionTwo"
          />
        </div>
        <div className="flex items-center justify-center">
          <button
            className="btn-accent"
            type="submit"
            disabled={newPoll.optionOne === "" || newPoll.optionTwo === ""}
          >
            Submit
          </button>
        </div>
      </form>
    </>
  );
};

export default AddPollForm;
