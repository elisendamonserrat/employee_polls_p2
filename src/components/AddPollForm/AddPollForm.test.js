/* eslint-disable testing-library/render-result-naming-convention */
/* eslint-disable testing-library/prefer-screen-queries */
import "@testing-library/jest-dom/extend-expect";
import { MemoryRouter } from "react-router-dom";
import { render, fireEvent, queryByAttribute } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../store";
import { setAuthedUser } from "../../actions/authedUser";

import AddPollForm from "./AddPollForm";

describe("PollForm", () => {
  it("won't match the snapshot when the user  types in the questions", async () => {
    // const users = await _getUsers();
    store.dispatch(setAuthedUser("sarahedo"));
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <AddPollForm />
        </Provider>
      </MemoryRouter>
    );
    const optionOneInput = component.getByTestId("optionOne");
    const optionTwoInput = component.getByTestId("optionTwo");
    fireEvent.change(optionOneInput, { target: { value: "Learning Jest" } });
    fireEvent.change(optionTwoInput, { target: { value: "Learning Mocha" } });
    expect(component).toMatchSnapshot();
  });
});
