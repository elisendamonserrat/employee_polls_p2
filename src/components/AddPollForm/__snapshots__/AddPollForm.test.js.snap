// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollForm won't match the snapshot when the user  types in the questions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="bg-white border shadow-md rounded px-8 pt-8 pb-8 mb-4 mt-8 text-left"
      >
        <p
          class="subtitle mb-4 font-bold"
        >
          Would You Rather...?
        </p>
        <div
          class="mb-4"
        >
          <label
            class="block text-sm font-bold mb-2"
            for="optionOne"
          >
            First Option
          </label>
          <input
            class="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
            data-testid="optionOne"
            id="optionOne"
            name="optionOne"
            placeholder="Write here your first option"
            type="text"
            value="Learning Jest"
          />
        </div>
        <div
          class="mb-6"
        >
          <label
            class="block text-sm font-bold mb-2"
            for="optionTwo"
          >
            Second Option
          </label>
          <input
            class="shadow appearance-none border rounded w-full py-2 px-3 mb-3 leading-tight focus:outline-none focus:shadow-outline"
            data-testid="optionTwo"
            id="optionTwo"
            name="optionTwo"
            placeholder="Write here your second option"
            type="text"
            value="Learning Mocha"
          />
        </div>
        <div
          class="flex items-center justify-center"
        >
          <button
            class="btn-accent"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="bg-white border shadow-md rounded px-8 pt-8 pb-8 mb-4 mt-8 text-left"
    >
      <p
        class="subtitle mb-4 font-bold"
      >
        Would You Rather...?
      </p>
      <div
        class="mb-4"
      >
        <label
          class="block text-sm font-bold mb-2"
          for="optionOne"
        >
          First Option
        </label>
        <input
          class="shadow appearance-none border rounded w-full py-2 px-3 leading-tight focus:outline-none focus:shadow-outline"
          data-testid="optionOne"
          id="optionOne"
          name="optionOne"
          placeholder="Write here your first option"
          type="text"
          value="Learning Jest"
        />
      </div>
      <div
        class="mb-6"
      >
        <label
          class="block text-sm font-bold mb-2"
          for="optionTwo"
        >
          Second Option
        </label>
        <input
          class="shadow appearance-none border rounded w-full py-2 px-3 mb-3 leading-tight focus:outline-none focus:shadow-outline"
          data-testid="optionTwo"
          id="optionTwo"
          name="optionTwo"
          placeholder="Write here your second option"
          type="text"
          value="Learning Mocha"
        />
      </div>
      <div
        class="flex items-center justify-center"
      >
        <button
          class="btn-accent"
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
